src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -285,43 +285,53 @@ export class FileStat implements IFileStat {
 	 * Returns a child stat from this stat that matches with the provided path.
 	 * Will return "null" in case the child does not exist.
 	 */
-	public find(resource: URI): FileStat {
-
+	public find(resource: URI): ExplorerItem {
 		// Return if path found
-		if (resources.isEqual(resource, this.resource, !isLinux /* ignorecase */)) {
-			return this;
+		if (resource && this.resource.scheme === resource.scheme && this.resource.authority === resource.authority &&
+			startsWith(resource.path, this.resource.path)
```

Comment:
@isidorn this checks seems to not work anymore if the path is the same but with different case? With our previous solution this would work. Also, given the code that is implemented in `findByPath` why do we actually need this extra check, isn't that covered by that other method already?